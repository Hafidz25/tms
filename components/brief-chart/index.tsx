import BriefChart from "./brief-chart";

export { BriefChart };
