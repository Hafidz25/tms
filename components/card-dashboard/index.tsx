import CardDashboard from "./card-dashboard";

export { CardDashboard };
