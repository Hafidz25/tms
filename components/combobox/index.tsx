export * from './combobox';
export { type ComboboxOption } from './use-combobox';