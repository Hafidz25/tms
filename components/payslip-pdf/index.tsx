import PayslipPdf from "./payslip-pdf";

export { PayslipPdf };
