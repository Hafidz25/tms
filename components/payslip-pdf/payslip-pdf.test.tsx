import { describe, expect, it, test } from "vitest";
import { render, screen, fireEvent } from "@testing-library/react";
import { PayslipPdf } from "@/components/payslip-pdf";

describe("Payslip PDF Component", () => {
  it("render payslip pdf component", () => {
    const component = render(<PayslipPdf />);
    expect(component).toMatchSnapshot();
  });
});
