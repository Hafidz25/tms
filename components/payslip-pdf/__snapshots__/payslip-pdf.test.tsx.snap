// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Payslip PDF Component > render payslip pdf component 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <document
        title="undefined - Payslip"
      >
        <page
          style="background-color: rgb(255, 255, 255); padding-top: 35px; padding-bottom: 65px;"
        >
          <view
            style="display: flex; flex-direction: row; align-items: flex-start; gap: 24px; height: 120px; border-bottom: 1.5px; border-bottom-color: #000; border-bottom-style: solid; margin-bottom: 30px;"
          >
            <image
              src="/payslip/logo.png"
              style="width: 94px; height: 94px;"
            />
            <view>
              <text
                style="font-family: Open Sans; font-size: 18px; font-weight: 700; margin-bottom: 12px;"
              >
                8AMProject Studio
              </text>
              <view
                style="display: flex; flex-direction: column; gap: 6px;"
              >
                <text
                  style="font-size: 12px;"
                >
                  Jalan Danau Sentani Utara I Blok H3 A5, Madyopuro, Kedungkandang,
                </text>
                <text
                  style="font-size: 12px;"
                >
                  Kota Malang Kode Pos 65139
                </text>
                <text
                  style="font-size: 12px;"
                >
                  Email : 8amproject.studio@gmail.com / www.8amproject.com
                </text>
              </view>
            </view>
          </view>
          <view
            style="display: flex; flex-direction: column; align-items: center; margin-bottom: 36px;"
          >
            <text
              style="font-size: 18px; text-decoration: underline; text-align: center; margin-bottom: 8px; font-family: Open Sans; font-weight: 700;"
            >
              SLIP GAJI KARYAWAN
            </text>
            <text
              style="font-size: 14px; text-align: center;"
            >
              Periode 
               - 
            </text>
          </view>
          <view
            style="display: flex; justify-content: flex-start; gap: 6px; margin-bottom: 28px;"
          >
            <text
              style="font-size: 12px;"
            >
              Nama       :  
            </text>
            <text
              style="font-size: 12px;"
            >
              Jabatan    :  
            </text>
            <text
              style="font-size: 12px;"
            >
              Level        :  
            </text>
          </view>
          <view
            style="margin-bottom: 12px;"
          >
            <text
              style="font-size: 14px; margin-bottom: 4px; font-family: Open Sans; font-weight: 700;"
            >
              PENGHASILAN
            </text>
            <view
              style="width: 100%; background-color: rgb(255, 193, 7); height: 36px; border: 2px; display: flex; flex-direction: row; margin-bottom: -1px;"
            >
              <view
                style="width: 12%; display: flex; align-items: center; justify-content: center; height: 34px; border-right: 1px;"
              >
                <text
                  style="font-size: 12px;"
                >
                  No.
                </text>
              </view>
              <view
                style="width: 53%; display: flex; align-items: center; justify-content: center; height: 34px; border-right: 1px;"
              >
                <text
                  style="font-size: 12px;"
                >
                  Keterangan
                </text>
              </view>
              <view
                style="width: 35%; display: flex; align-items: center; justify-content: center; height: 34px;"
              >
                <text
                  style="font-size: 12px;"
                >
                  Nominal
                </text>
              </view>
            </view>
            <view
              style="width: 100%; background-color: rgb(255, 255, 255); height: 36px; border-left: 2px; border-right: 2px; border-bottom: 2px; display: flex; flex-direction: row; margin-bottom: -1px;"
            >
              <view
                style="width: 12%; display: flex; align-items: center; justify-content: center; height: 34px; border-right: 1px;"
              >
                <text
                  style="font-size: 12px;"
                >
                  1.
                </text>
              </view>
              <view
                style="width: 53%; display: flex; align-items: flex-start; justify-content: center; height: 34px; border-right: 1px;"
              >
                <text
                  style="font-size: 12px;"
                >
                  Gaji Pokok
                </text>
              </view>
              <view
                style="width: 35%; display: flex; align-items: flex-start; justify-content: center; background-color: rgb(255, 229, 152); height: 34px; border-bottom: 0.5px;"
              >
                <text
                  style="font-size: 12px;"
                >
                  RpNaN
                </text>
              </view>
            </view>
            <view
              style="width: 100%; background-color: rgb(255, 255, 255); height: 36px; border-left: 2px; border-right: 2px; border-bottom: 2px; display: flex; flex-direction: row; margin-bottom: -1px;"
            >
              <view
                style="width: 12%; display: flex; align-items: center; justify-content: center; height: 34px; border-right: 1px;"
              >
                <text
                  style="font-size: 12px;"
                >
                  2.
                </text>
              </view>
              <view
                style="width: 53%; display: flex; align-items: flex-start; justify-content: center; height: 34px; border-right: 1px;"
              >
                <text
                  style="font-size: 12px;"
                >
                  Tunjangan Transportasi (
                  )
                </text>
              </view>
              <view
                style="width: 35%; display: flex; align-items: flex-start; justify-content: center; background-color: rgb(255, 229, 152); height: 34px; border-bottom: 0.5px;"
              >
                <text
                  style="font-size: 12px;"
                >
                  RpNaN
                </text>
              </view>
            </view>
            <view
              style="width: 100%; background-color: rgb(255, 255, 255); height: 36px; border-left: 2px; border-right: 2px; border-bottom: 2px; display: flex; flex-direction: row; margin-bottom: -1px;"
            >
              <view
                style="width: 12%; display: flex; align-items: center; justify-content: center; height: 34px; border-right: 1px;"
              >
                <text
                  style="font-size: 12px;"
                >
                  3.
                </text>
              </view>
              <view
                style="width: 53%; display: flex; align-items: flex-start; justify-content: center; height: 34px; border-right: 1px;"
              >
                <text
                  style="font-size: 12px;"
                >
                  THR
                </text>
              </view>
              <view
                style="width: 35%; display: flex; align-items: flex-start; justify-content: center; background-color: rgb(255, 229, 152); height: 34px; border-bottom: 0.5px;"
              >
                <text
                  style="font-size: 12px;"
                >
                  RpNaN
                </text>
              </view>
            </view>
            <view
              style="width: 100%; background-color: rgb(255, 255, 255); height: 36px; border-left: 2px; border-right: 2px; border-bottom: 2px; display: flex; flex-direction: row; margin-bottom: -1px;"
            >
              <view
                style="width: 12%; display: flex; align-items: center; justify-content: center; height: 34px; border-right: 1px;"
              >
                <text
                  style="font-size: 12px;"
                >
                  4.
                </text>
              </view>
              <view
                style="width: 53%; display: flex; align-items: flex-start; justify-content: center; height: 34px; border-right: 1px;"
              >
                <text
                  style="font-size: 12px;"
                >
                  Lain - Lain
                </text>
              </view>
              <view
                style="width: 35%; display: flex; align-items: flex-start; justify-content: center; background-color: rgb(255, 229, 152); height: 34px; border-bottom: 0.5px;"
              >
                <text
                  style="font-size: 12px;"
                >
                  RpNaN
                </text>
              </view>
            </view>
            <view
              style="width: 100%; background-color: rgb(255, 255, 255); height: 36px; border-left: 2px; border-right: 2px; border-bottom: 2px; display: flex; flex-direction: row; margin-bottom: -1px;"
            >
              <view
                style="width: 65%; display: flex; align-items: center; justify-content: center; height: 34px; border-right: 1px;"
              >
                <text
                  style="font-size: 12px;"
                >
                  TOTAL
                </text>
              </view>
              <view
                style="width: 35%; display: flex; align-items: flex-start; justify-content: center; background-color: rgb(255, 229, 152); height: 34px; border-bottom: 0.5px;"
              >
                <text
                  style="font-size: 12px;"
                >
                  RpNaN
                </text>
              </view>
            </view>
          </view>
          <view
            style="display: flex; flex-direction: column; gap: 6px; margin-bottom: 36px;"
          >
            <text
              style="font-size: 12px;"
            >
              Keterangan :
            </text>
            <view
              style="display: flex; flex-direction: row; gap: 6px; align-items: flex-start; margin-left: 12px;"
            >
              <text
                style="font-size: 12px;"
              >
                1.
              </text>
              <text
                style="font-size: 12px;"
              >
                THR untuk &gt; 1 Tahun, Senilai 1 Kali Gaji Pokok
              </text>
            </view>
            <view
              style="display: flex; flex-direction: row; gap: 6px; align-items: flex-start; margin-left: 12px;"
            >
              <text
                style="font-size: 12px;"
              >
                2.
              </text>
              <text
                style="font-size: 12px;"
              >
                Gaji Dibayarkan setiap Tanggal 1-3 bulan berikutnya
              </text>
            </view>
            <view
              style="display: flex; flex-direction: row; gap: 6px; align-items: flex-start; margin-left: 12px;"
            >
              <text
                style="font-size: 12px;"
              >
                3.
              </text>
              <text
                style="font-size: 12px;"
              >
                Apabila terjadi kesalahan perhitungan laporkan sebelum tgl 1 setiap bulannya
              </text>
            </view>
            <view
              style="display: flex; flex-direction: row; gap: 6px; align-items: flex-start; margin-left: 12px;"
            >
              <text
                style="font-size: 12px;"
              >
                4.
              </text>
              <text
                style="font-size: 12px;"
              >
                Kenaikan Level akan ditentukan setelah masa Kontrak 1 tahun. Level akan Direview setiap tahunnya sesuai dengan performa dan diinformasikan ke masing-masing team
              </text>
            </view>
          </view>
          <view
            style="display: flex; flex-direction: column; gap: 2px; align-items: flex-end;"
          >
            <text
              style="font-size: 12px;"
            >
              Malang, 
              06 Agustus 2024
            </text>
            <image
              src="/payslip/signature.jpg"
              style="width: 50px;"
            />
            <text
              style="font-size: 12px;"
            >
              Robi Wahyudi
            </text>
            <text
              style="font-size: 12px;"
            >
              8AMProject Studio
            </text>
          </view>
        </page>
      </document>
    </div>
  </body>,
  "container": <div>
    <document
      title="undefined - Payslip"
    >
      <page
        style="background-color: rgb(255, 255, 255); padding-top: 35px; padding-bottom: 65px;"
      >
        <view
          style="display: flex; flex-direction: row; align-items: flex-start; gap: 24px; height: 120px; border-bottom: 1.5px; border-bottom-color: #000; border-bottom-style: solid; margin-bottom: 30px;"
        >
          <image
            src="/payslip/logo.png"
            style="width: 94px; height: 94px;"
          />
          <view>
            <text
              style="font-family: Open Sans; font-size: 18px; font-weight: 700; margin-bottom: 12px;"
            >
              8AMProject Studio
            </text>
            <view
              style="display: flex; flex-direction: column; gap: 6px;"
            >
              <text
                style="font-size: 12px;"
              >
                Jalan Danau Sentani Utara I Blok H3 A5, Madyopuro, Kedungkandang,
              </text>
              <text
                style="font-size: 12px;"
              >
                Kota Malang Kode Pos 65139
              </text>
              <text
                style="font-size: 12px;"
              >
                Email : 8amproject.studio@gmail.com / www.8amproject.com
              </text>
            </view>
          </view>
        </view>
        <view
          style="display: flex; flex-direction: column; align-items: center; margin-bottom: 36px;"
        >
          <text
            style="font-size: 18px; text-decoration: underline; text-align: center; margin-bottom: 8px; font-family: Open Sans; font-weight: 700;"
          >
            SLIP GAJI KARYAWAN
          </text>
          <text
            style="font-size: 14px; text-align: center;"
          >
            Periode 
             - 
          </text>
        </view>
        <view
          style="display: flex; justify-content: flex-start; gap: 6px; margin-bottom: 28px;"
        >
          <text
            style="font-size: 12px;"
          >
            Nama       :  
          </text>
          <text
            style="font-size: 12px;"
          >
            Jabatan    :  
          </text>
          <text
            style="font-size: 12px;"
          >
            Level        :  
          </text>
        </view>
        <view
          style="margin-bottom: 12px;"
        >
          <text
            style="font-size: 14px; margin-bottom: 4px; font-family: Open Sans; font-weight: 700;"
          >
            PENGHASILAN
          </text>
          <view
            style="width: 100%; background-color: rgb(255, 193, 7); height: 36px; border: 2px; display: flex; flex-direction: row; margin-bottom: -1px;"
          >
            <view
              style="width: 12%; display: flex; align-items: center; justify-content: center; height: 34px; border-right: 1px;"
            >
              <text
                style="font-size: 12px;"
              >
                No.
              </text>
            </view>
            <view
              style="width: 53%; display: flex; align-items: center; justify-content: center; height: 34px; border-right: 1px;"
            >
              <text
                style="font-size: 12px;"
              >
                Keterangan
              </text>
            </view>
            <view
              style="width: 35%; display: flex; align-items: center; justify-content: center; height: 34px;"
            >
              <text
                style="font-size: 12px;"
              >
                Nominal
              </text>
            </view>
          </view>
          <view
            style="width: 100%; background-color: rgb(255, 255, 255); height: 36px; border-left: 2px; border-right: 2px; border-bottom: 2px; display: flex; flex-direction: row; margin-bottom: -1px;"
          >
            <view
              style="width: 12%; display: flex; align-items: center; justify-content: center; height: 34px; border-right: 1px;"
            >
              <text
                style="font-size: 12px;"
              >
                1.
              </text>
            </view>
            <view
              style="width: 53%; display: flex; align-items: flex-start; justify-content: center; height: 34px; border-right: 1px;"
            >
              <text
                style="font-size: 12px;"
              >
                Gaji Pokok
              </text>
            </view>
            <view
              style="width: 35%; display: flex; align-items: flex-start; justify-content: center; background-color: rgb(255, 229, 152); height: 34px; border-bottom: 0.5px;"
            >
              <text
                style="font-size: 12px;"
              >
                RpNaN
              </text>
            </view>
          </view>
          <view
            style="width: 100%; background-color: rgb(255, 255, 255); height: 36px; border-left: 2px; border-right: 2px; border-bottom: 2px; display: flex; flex-direction: row; margin-bottom: -1px;"
          >
            <view
              style="width: 12%; display: flex; align-items: center; justify-content: center; height: 34px; border-right: 1px;"
            >
              <text
                style="font-size: 12px;"
              >
                2.
              </text>
            </view>
            <view
              style="width: 53%; display: flex; align-items: flex-start; justify-content: center; height: 34px; border-right: 1px;"
            >
              <text
                style="font-size: 12px;"
              >
                Tunjangan Transportasi (
                )
              </text>
            </view>
            <view
              style="width: 35%; display: flex; align-items: flex-start; justify-content: center; background-color: rgb(255, 229, 152); height: 34px; border-bottom: 0.5px;"
            >
              <text
                style="font-size: 12px;"
              >
                RpNaN
              </text>
            </view>
          </view>
          <view
            style="width: 100%; background-color: rgb(255, 255, 255); height: 36px; border-left: 2px; border-right: 2px; border-bottom: 2px; display: flex; flex-direction: row; margin-bottom: -1px;"
          >
            <view
              style="width: 12%; display: flex; align-items: center; justify-content: center; height: 34px; border-right: 1px;"
            >
              <text
                style="font-size: 12px;"
              >
                3.
              </text>
            </view>
            <view
              style="width: 53%; display: flex; align-items: flex-start; justify-content: center; height: 34px; border-right: 1px;"
            >
              <text
                style="font-size: 12px;"
              >
                THR
              </text>
            </view>
            <view
              style="width: 35%; display: flex; align-items: flex-start; justify-content: center; background-color: rgb(255, 229, 152); height: 34px; border-bottom: 0.5px;"
            >
              <text
                style="font-size: 12px;"
              >
                RpNaN
              </text>
            </view>
          </view>
          <view
            style="width: 100%; background-color: rgb(255, 255, 255); height: 36px; border-left: 2px; border-right: 2px; border-bottom: 2px; display: flex; flex-direction: row; margin-bottom: -1px;"
          >
            <view
              style="width: 12%; display: flex; align-items: center; justify-content: center; height: 34px; border-right: 1px;"
            >
              <text
                style="font-size: 12px;"
              >
                4.
              </text>
            </view>
            <view
              style="width: 53%; display: flex; align-items: flex-start; justify-content: center; height: 34px; border-right: 1px;"
            >
              <text
                style="font-size: 12px;"
              >
                Lain - Lain
              </text>
            </view>
            <view
              style="width: 35%; display: flex; align-items: flex-start; justify-content: center; background-color: rgb(255, 229, 152); height: 34px; border-bottom: 0.5px;"
            >
              <text
                style="font-size: 12px;"
              >
                RpNaN
              </text>
            </view>
          </view>
          <view
            style="width: 100%; background-color: rgb(255, 255, 255); height: 36px; border-left: 2px; border-right: 2px; border-bottom: 2px; display: flex; flex-direction: row; margin-bottom: -1px;"
          >
            <view
              style="width: 65%; display: flex; align-items: center; justify-content: center; height: 34px; border-right: 1px;"
            >
              <text
                style="font-size: 12px;"
              >
                TOTAL
              </text>
            </view>
            <view
              style="width: 35%; display: flex; align-items: flex-start; justify-content: center; background-color: rgb(255, 229, 152); height: 34px; border-bottom: 0.5px;"
            >
              <text
                style="font-size: 12px;"
              >
                RpNaN
              </text>
            </view>
          </view>
        </view>
        <view
          style="display: flex; flex-direction: column; gap: 6px; margin-bottom: 36px;"
        >
          <text
            style="font-size: 12px;"
          >
            Keterangan :
          </text>
          <view
            style="display: flex; flex-direction: row; gap: 6px; align-items: flex-start; margin-left: 12px;"
          >
            <text
              style="font-size: 12px;"
            >
              1.
            </text>
            <text
              style="font-size: 12px;"
            >
              THR untuk &gt; 1 Tahun, Senilai 1 Kali Gaji Pokok
            </text>
          </view>
          <view
            style="display: flex; flex-direction: row; gap: 6px; align-items: flex-start; margin-left: 12px;"
          >
            <text
              style="font-size: 12px;"
            >
              2.
            </text>
            <text
              style="font-size: 12px;"
            >
              Gaji Dibayarkan setiap Tanggal 1-3 bulan berikutnya
            </text>
          </view>
          <view
            style="display: flex; flex-direction: row; gap: 6px; align-items: flex-start; margin-left: 12px;"
          >
            <text
              style="font-size: 12px;"
            >
              3.
            </text>
            <text
              style="font-size: 12px;"
            >
              Apabila terjadi kesalahan perhitungan laporkan sebelum tgl 1 setiap bulannya
            </text>
          </view>
          <view
            style="display: flex; flex-direction: row; gap: 6px; align-items: flex-start; margin-left: 12px;"
          >
            <text
              style="font-size: 12px;"
            >
              4.
            </text>
            <text
              style="font-size: 12px;"
            >
              Kenaikan Level akan ditentukan setelah masa Kontrak 1 tahun. Level akan Direview setiap tahunnya sesuai dengan performa dan diinformasikan ke masing-masing team
            </text>
          </view>
        </view>
        <view
          style="display: flex; flex-direction: column; gap: 2px; align-items: flex-end;"
        >
          <text
            style="font-size: 12px;"
          >
            Malang, 
            06 Agustus 2024
          </text>
          <image
            src="/payslip/signature.jpg"
            style="width: 50px;"
          />
          <text
            style="font-size: 12px;"
          >
            Robi Wahyudi
          </text>
          <text
            style="font-size: 12px;"
          >
            8AMProject Studio
          </text>
        </view>
      </page>
    </document>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
